### PR TITLE
[Temporarily] add required/desired NeXus metadata for i22 detectors

### PR DESCRIPTION
**Pull request by DiamondJoseph**
Required for i22 June experiment

### Instructions to reviewer on how to test:
1. check that i22 saxs/waxs/oav have expected field in describe_configuration(), describe()

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

**Comment by DiamondJoseph** [comment]
Requires this change https://github.com/bluesky/ophyd-async/pull/363 to allow soft_signal to have units.

File: src/dodal/devices/i22/nxsas.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                self.serial_number = None
+            if sensor_thickness is not None:
+                self.sensor_thickness, _ = soft_signal_r_and_setter(
+                    float, sensor_thickness[0], units=sensor_thickness[1]
+                )
+            else:
+                self.sensor_thickness = None
+            if threshold_energy is not None:
+                self.threshold_energy, _ = soft_signal_r_and_setter(
+                    float, threshold_energy[0], threshold_energy[1]
```
Comment (DiamondJoseph) [inline src/dodal/devices/i22/nxsas.py:111]:
```suggestion
                    float, threshold_energy[0], units=threshold_energy[1]
```

File: src/dodal/devices/i22/nxsas.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                self.azimuthal_angle = None
+            if rotation_angle is not None:
+                self.rotation_angle, _ = soft_signal_r_and_setter(
+                    float, rotation_angle[0], units=rotation_angle[1]
+                )
+            else:
+                self.rotation_angle = None
+            if aequetorial_angle is not None:
+                self.aequetorial_angle, _ = soft_signal_r_and_setter(
+                    float, aequetorial_angle[0], aequetorial_angle[1]
```
Comment (DiamondJoseph) [inline src/dodal/devices/i22/nxsas.py:71]:
```suggestion
                    float, aequetorial_angle[0], units=aequetorial_angle[1]
```

File: src/dodal/devices/i22/nxsas.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                    float, azimuthal_angle[0], units=azimuthal_angle[1]
+                )
+            else:
+                self.azimuthal_angle = None
+            if rotation_angle is not None:
+                self.rotation_angle, _ = soft_signal_r_and_setter(
+                    float, rotation_angle[0], units=rotation_angle[1]
+                )
+            else:
+                self.rotation_angle = None
```
Comment (stan-dot) [inline src/dodal/devices/i22/nxsas.py:68]: make this a oneliner: `self.rotation_angle, _ = soft_signal_r_and_setter(float, rotation_angle[0], units=rotation_angle[1]) if rotation_angle is not None else (None, None)
`

Comment (stan-dot) [reply src/dodal/devices/i22/nxsas.py:68]: and others too

File: src/dodal/devices/i22/nxsas.py [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+                self.threshold_energy = None
+
+
+class NXSasPilatus(PilatusDetector):
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        drv_suffix,
+        hdf_suffix,
```
Comment (stan-dot) [inline src/dodal/devices/i22/nxsas.py:123]: those miss the type

File: src/dodal/devices/i22/nxsas.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            hdf_suffix=hdf_suffix,
+            name=name,
+        )
+        self._metadata_holder = metadata_holder
+
+    async def describe_configuration(self) -> Dict[str, DataKey]:
+        return {
+            **await super().describe_configuration(),
+            **await self._metadata_holder.describe_configuration(),
+        }
```
Comment (stan-dot) [inline src/dodal/devices/i22/nxsas.py:140]: pls add a comment what is going on here and why do we need to extra metadata

File: src/dodal/devices/i22/nxsas.py [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+
+
+class NXSasOAV(AravisDetector):
+    def __init__(
+        self,
+        prefix: str,
+        directory_provider: DirectoryProvider,
+        drv_suffix,
+        hdf_suffix,
```
Comment (stan-dot) [inline src/dodal/devices/i22/nxsas.py:155]: ditto: add str

**Review by stan-dot — CHANGES_REQUESTED** [review]

File: src/dodal/devices/i22/nxsas.py (hunk 3 above)
Comment (DiamondJoseph) [reply src/dodal/devices/i22/nxsas.py:68]: This is an established pattern for soft signals for the moment.
I agree that it's verbose, but unfortunately it seems to be required for type checking
https://github.com/DiamondLightSource/dodal/blob/0e15a5ffd5bf5b8b9ed09a81c6d37608af0272c9/src/dodal/devices/i22/fswitch.py#L54

File: src/dodal/devices/i22/nxsas.py (hunk 5 above)
Comment (DiamondJoseph) [reply src/dodal/devices/i22/nxsas.py:146]: I've added docstrings to the type, is that sufficient?

File: src/dodal/devices/i22/nxsas.py (hunk 3 above)
Comment (stan-dot) [reply src/dodal/devices/i22/nxsas.py:69]: ok I see, I didn't realize that.

**Review by stan-dot — APPROVED** [review]
all good

**Pull request merged**